core/chains/evm/txmgr/models.go

```diff
@@ -160,11 +178,12 @@ func (e *NullableEIP2930AccessList) Scan(value interface{}) error {
 	}
 }
 
-type EthTx struct {
+type EthTx[ADDR commontypes.Hashable, TX_HASH commontypes.Hashable] struct {
+	txmgrtypes.Transaction
```

Comment:
actually on giving it more thought - please do this. it would save a lot of extraneous lines in this PR since `Tx` becomes the new base type and `EthTx` remains the evm specific type and then we don't need to change all the test files over and over

Reply:
it would look something like

```go
type Tx[ADDR commontypes.Hashable, TX_HASH commontyupes.Hashable] struct {
	...
}

type EthTx Tx[evmtypes.Address, evmtypes.Hash]
```

Reply:
Hmm, didn't fully understand.
We already have this defined:
`type EvmTx = EthTx[*evmtypes.Address, *evmtypes.TxHash]`
EvmTx is now already used in all places.
Did you just mean to rename EvmTx as EthTx, so we wouldn't have to change other places? I think eventually, we would still want to change all Eth to Evm for correctness sake.

I actually also intend to change the base type to Tx(or equiv) and also move it to common.txmgr package, in another PR.